src/requestHandlers/hook.ts

```diff
@@ -0,0 +1,150 @@
+import { Request, Response } from 'express';
+import * as hookService from '../services/hookService';
+import { assertArguments, assertArgumentsDefined, sanitizeResponse } from './utils';
+import { getPanel } from '../services/panelService';
+import { getPanelSetByID } from '../services/panelSetService';
+import { sequelize } from '../database';
+import { Sequelize } from 'sequelize';
+
+
+/**
+ * Create a hook
+ * @param {number[]} position Hook position on panel
+ * @param {number} current_panel_id ID of panel hook is on
+ * @param {number} next_panel_set_id ID of panel set that hook links to
+ * @returns response or error
+ */
+const _createHookController = (sequelize: Sequelize) => async (position: number[], current_panel_id: number, next_panel_set_id: number) => {
+    try {
+        const panel = await getPanel(sequelize)(current_panel_id);
+        if (panel == null) throw new Error('no panel exists for given panel id');
+        return await hookService.createHook(sequelize)({
+            position,
+            current_panel_id,
+            next_panel_set_id
+        });
+    }
+    catch (err) {
+        return err;
+    }
+};
+
+/**
+ * Create Hook request
+ * @param {Request} req 
+ * @param {Response} res 
+ * @returns 
+ */
+const createHook = async (req: Request, res: Response): Promise<Response> => {
+    const position : number[] = req.body.position;
+    const current_panel_id : number = req.body.current_panel_id;
+    const next_panel_set_id : number = req.body.next_panel_set_id;
+
+    const validArgs = assertArguments(
+        { position, current_panel_id, next_panel_set_id },
+        a => a != undefined || a === null,
```

Comment:
`next_panel_set_id` can (and often will) equal `null` so it is important that `null` specifically be a valid argument. That is why I don't use `assertArgumentsDefined`. Not sure if there was a more elegant way to go about this

Reply:
If you want to assert `a !== undefined` but allow null, then that's fine I'll approve the PR.

Otherwise, I would just leave out `next_panel_set_id` from the validator since it's allowed to be null/undefined.

Reply:
Left it out, good idea